checkQC/app.py

```diff
@@ -1,27 +1,54 @@
 
 import sys
 
+import click
+
 from checkQC.qc_engine import QCEngine
-from checkQC.config import get_config
+from checkQC.config import get_config, get_handler_config
 from checkQC.run_type_recognizer import RunTypeRecognizer
 
 
-def start():
+@click.command("checkQC")
+@click.option("--config_file", help="Path to the checkQC configuration file", type=click.Path())
+@click.argument('runfolder', type=click.Path())
+def start(config_file, runfolder):
+    """
+    checkQC is a command line utility designed to quickly gather and assess quality control metrics from a
+    Illumina sequencing run. It is highly customizable and which quality controls modules should be run
+    for a particular run type should be specified in the provided configuration file.
+    """
+    # -----------------------------------
+    # This is the application entry point
+    # -----------------------------------
+
+    app = App(runfolder, config_file)
+    app.run()
+    sys.exit(app.exit_status)
+
+
+class App(object):
 
-    config = get_config("config/config.yaml")
+    def __init__(self, runfolder, config_file=None):
+        self._runfolder = runfolder
+        self._config_file = config_file
+        self.exit_status = 0
 
-    runfolder = "./tests/resources/MiSeqDemo"
+    def run(self):
+        config = get_config(self._config_file)
 
-    run_type_recognizer = RunTypeRecognizer(config=config, runfolder=runfolder)
-    instrument_type = run_type_recognizer.instrument_type()
-    read_length = run_type_recognizer.read_length()
+        run_type_recognizer = RunTypeRecognizer(config=config, runfolder=self._runfolder)
+        instrument_type = run_type_recognizer.instrument_type()
+        reagent_version = run_type_recognizer.reagent_version()
 
-    handler_config = config[instrument_type][read_length]["handlers"]
+        # TODO For now assume symmetric read lengths
+        read_length = int(run_type_recognizer.read_length().split("-")[0])
```

Comment:
Would it make sense for the recognizer to return the read_length? Then only it knows the format.